libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -206,16 +204,17 @@ private void initNavigationLocationEngine() {
     locationEngineConductor = new LocationEngineConductor(locationEngineCallback);
   }
 
-  private void initLanguage(NavigationUiOptions options, LocaleUtils localeUtils) {
-    language = localeUtils.getNonEmptyLanguage(getApplication(), options.directionsRoute().voiceLanguage());
+  private void initLanguage(NavigationUiOptions options) {
+    language = options.directionsRoute().routeOptions().language();
   }
 
-  private void initUnitType(NavigationUiOptions options, LocaleUtils localeUtils) {
+  private void initUnitType(NavigationUiOptions options) {
     unitType = options.directionsRoute().routeOptions().voiceUnits();
   }
 
-  private void initVoiceInstructions() {
-    instructionPlayer = new NavigationInstructionPlayer(getApplication(), language, accessToken);
+  private void initVoiceInstructions(NavigationViewOptions options) {
```

Comment:
Same here ☝️

Reply:
We do this a lot of places (pass in the options to initialize something), I think even just for consistency it should stay how it is

Reply:
Yeah I was thinking on updating all of them 😅

Reply:
@Guardiola31337 @danesfeder  Maybe we should all talk about that and come to a consensus so we know what to do going forward